satpy/readers/olci_nc.py

```diff
@@ -319,67 +307,62 @@ class NCOLCIAngles(NCOLCILowResData):
-        if key['name'] not in self.datasets:
+        key_name = key['name']
+        if key_name not in self.datasets:
             return
 
-        self._open_dataset()
+        logger.debug('Reading %s.', key_name)
 
-        logger.debug('Reading %s.', key['name'])
-
-        if self._need_interpolation() and self.cache.get(key['name']) is None:
-
-            if key['name'].startswith('satellite'):
-                zen = self.nc[self.datasets['satellite_zenith_angle']]
-                zattrs = zen.attrs
-                azi = self.nc[self.datasets['satellite_azimuth_angle']]
-                aattrs = azi.attrs
-            elif key['name'].startswith('solar'):
-                zen = self.nc[self.datasets['solar_zenith_angle']]
-                zattrs = zen.attrs
-                azi = self.nc[self.datasets['solar_azimuth_angle']]
-                aattrs = azi.attrs
-            else:
-                raise NotImplementedError("Don't know how to read " + key['name'])
-
-            x, y, z = angle2xyz(azi, zen)
-
-            x, y, z = self._do_interpolate((x, y, z))
-
-            azi, zen = xyz2angle(x, y, z)
-            azi.attrs = aattrs
-            zen.attrs = zattrs
-
-            if 'zenith' in key['name']:
-                values = zen
-            elif 'azimuth' in key['name']:
-                values = azi
-            else:
-                raise NotImplementedError("Don't know how to read " + key['name'])
-
-            if key['name'].startswith('satellite'):
-                self.cache['satellite_zenith_angle'] = zen
-                self.cache['satellite_azimuth_angle'] = azi
-            elif key['name'].startswith('solar'):
-                self.cache['solar_zenith_angle'] = zen
-                self.cache['solar_azimuth_angle'] = azi
-
-        elif key['name'] in self.cache:
-            values = self.cache[key['name']]
+        if self._needs_interpolation():
+            data = self._get_interpolated_dataset(key_name)
         else:
-            values = self.nc[self.datasets[key['name']]]
-
-        values.attrs['platform_name'] = self.platform_name
-        values.attrs['sensor'] = self.sensor
-
-        values.attrs.update(key.to_dict())
-        return values
-
-    def __del__(self):
-        """Close the NetCDF file that may still be open."""
-        try:
-            self.nc.close()
-        except (OSError, AttributeError):
-            pass
+            data = self.nc[self.datasets[key_name]]
+
+        self._fill_dataarray_attrs(data, key)
+        return data
+
+    def _get_interpolated_dataset(self, key_name):
+        """Get the interpolated dataset."""
+        azi_key, zen_key = self._get_angle_dataset_names(key_name)
+        azi, zen = self._get_full_resolution_angles(azi_key, zen_key)
+        if 'zenith' in key_name:
+            data = zen
+        elif 'azimuth' in key_name:
+            data = azi
+        else:
+            raise NotImplementedError("Don't know how to read " + key_name)
```

Comment:
Should this be a ValueError or RuntimeError? Something more likely to be caught by code calling this.

Reply:
Same for the one just below it.

Reply:
I was actually considering removing this error, since then only time this would be used is if someone changes the yaml file to add a new dataset and forgot to change in the code. Would you be ok with me removing this @djhoese ?

Reply:
Makes sense...except then do you just have an `elif` with no else and nothing after it? Don't linters hate that?

Reply:
I'll have a look